composer/loggers/logger_destination.py

```diff
@@ -137,3 +137,22 @@ def log_symlink_artifact(
         """
         del state, log_level, existing_artifact_name, symlink_artifact_name, overwrite  # unused
         pass
+
+    def get_file_artifact(
+        self,
+        artifact_name: str,
+        destination: str,
+        chunk_size: int = 2**20,
+        progress_bar: bool = True,
+    ):
+        """Handle downloading an artifact named ``artifact_name`` to ``destination``.
+
+        Args:
+            artifact_name (str): The name of the artifact.
+            destination (str): The destination filepath.
+            chunk_size (int, optional): Chunk size (in bytes). Ignored if ``path`` is a local file. (default: 1MB)
+            progress_bar (bool, optional): Whether to show a progress bar. Ignored if ``path`` is a local file.
+                (default: ``True``)
+        """
+        del artifact_name, destination, chunk_size, progress_bar  # unused
+        pass
```

Comment:
Can you raise a `NotImplementedError` here? Unlike the other APIs (which give the logger the choice on whether to log something), I think it is important that the caller know if the file isn't downloaded.